src/SonarScanner.MSBuild.TFS/CoverageReportProcessorBase.cs

```diff
@@ -93,25 +94,28 @@ public bool ProcessCoverageReports()
         private static bool IsVsTestReportsPaths(Property property) =>
             Property.AreKeysEqual(property.Id, SonarProperties.VsTestReportsPaths);
 
-        protected abstract bool TryGetBinaryReportFile(AnalysisConfig config, ITeamBuildSettings settings, out string binaryFilePath);
+        protected abstract bool TryGetVsCoverageFiles(AnalysisConfig config, ITeamBuildSettings settings, out IEnumerable<string> binaryFilePaths);
 
-        protected abstract bool TryGetTrxFile(AnalysisConfig config, ITeamBuildSettings settings, out string trxFilePath);
+        protected abstract bool TryGetTrxFiles(AnalysisConfig config, ITeamBuildSettings settings, out IEnumerable<string> trxFilePaths);
 
-        private bool TryConvertCoverageReport(string binaryCoverageFilePath, out string coverageReportFileName)
+        private bool TryConvertCoverageReports(IEnumerable<string> vscoverageFilePaths, out IEnumerable<string> vscoveragexmlPaths)
         {
-            coverageReportFileName = null;
-            var xmlFileName = Path.ChangeExtension(binaryCoverageFilePath, XmlReportFileExtension);
+            var xmlFileNames = vscoverageFilePaths.Select(x => Path.ChangeExtension(x, XmlReportFileExtension));
 
-            Debug.Assert(!File.Exists(xmlFileName),
-                "Not expecting a file with the name of the binary-to-XML conversion output to already exist: " + xmlFileName);
+            Debug.Assert(!xmlFileNames.Any(x => File.Exists(x)),
+                "Not expecting a file with the name of the binary-to-XML conversion output to already exist.");
 
-            if (converter.ConvertToXml(binaryCoverageFilePath, xmlFileName))
+            var anyFailedConversion = vscoverageFilePaths.Zip(xmlFileNames, (x, y) => new { coverage = x, xml = y })
```

Comment:
I found this method unnecessary confusing and hard to follow. Rather than creating a second list then zipping it to the first list, it would be simpler to create the combined list in original select in line 103.
Another simpler option would just have been to write a _for-each_ loop over _vscoveragefilePaths_ (which would also have few object allocations, although that level of optimisation isn't important here).